tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -42,26 +43,287 @@ limitations under the License.
+                      *dst_md_, strides_, dilations,
+                      padding_l_, padding_r_, padding_kind::zero));
+    }
+
+    fwd_pd_.reset(new convolution_forward::primitive_desc(
+                  *fwd_desc_, cpu_engine_));
+
+    // store the expected memory format
+    src_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->src_primitive_desc().desc().data.format);
+
+    filter_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->weights_primitive_desc().desc().data.format);
+    dst_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->dst_primitive_desc().desc().data.format);
+
+    // create memory primitive based on dummy data
+    src_mem_.reset(new memory(fwd_pd_.get()->src_primitive_desc(), DummyData));
+    filter_mem_.reset(new memory(fwd_pd_.get()->weights_primitive_desc(),
+                      DummyData));
+    dst_mem_.reset(new memory(fwd_pd_.get()->dst_primitive_desc(), DummyData));
+
+    // create convolution primitive and add it to net
+    if (!bias_dims.empty()) {
+        bias_mem_.reset(new memory({{{bias_dims}, MklDnnType<T>(),
+                             memory::format::x}, cpu_engine_}, DummyData));
+        conv_fwd_.reset(new convolution_forward(*fwd_pd_, *src_mem_,
+                        *filter_mem_, *bias_mem_, *dst_mem_));
+    } else {
+        conv_fwd_.reset(new convolution_forward(*fwd_pd_, *src_mem_,
+                        *filter_mem_, *dst_mem_));
+    }
+
+    fwd_primitives_.push_back(*conv_fwd_);
+    return;
+  }
+
+ public:
+  // Convolution forward execute with bias
+  void Execute(void* src, void* w, void* b, void* dst) {
+    src_mem_->set_data_handle(src);
+    filter_mem_->set_data_handle(w);
+    bias_mem_->set_data_handle(b);
+    dst_mem_->set_data_handle(dst);
+    fwd_stream_->submit(fwd_primitives_);
+
+    // after exec, set data handle back
+    src_mem_->set_data_handle(DummyData);
+    filter_mem_->set_data_handle(DummyData);
+    bias_mem_->set_data_handle(DummyData);
+    dst_mem_->set_data_handle(DummyData);
+
+    return;
+  }
+
+  // Convolution forward execute without bias
+  void Execute(void* src, void* w, void* dst) {
+    src_mem_->set_data_handle(src);
+    filter_mem_->set_data_handle(w);
+    dst_mem_->set_data_handle(dst);
+    fwd_stream_->submit(fwd_primitives_);
+
+    // after exec, set data handle back
+    src_mem_->set_data_handle(DummyData);
+    filter_mem_->set_data_handle(DummyData);
+    dst_mem_->set_data_handle(DummyData);
+
+    return;
+  }
+
+  // expected memory format for this primitive instance
+  memory::format src_fmt_;
+  memory::format filter_fmt_;
+  memory::format dst_fmt_;
+
+  // convolution primitive
+  std::shared_ptr<mkldnn::convolution_forward::primitive_desc> fwd_pd_;
+  std::shared_ptr<mkldnn::primitive> conv_fwd_;
+
+ private:
+  // MKLDNN memory
+  std::shared_ptr<mkldnn::memory> src_mem_;
```

Comment:
shared_ptr has been commonly used by the open source MKL-DNN. So we follow up with the same style. If required, we can replaced it with unique_ptr. 
Using shared_ptr seems necessary as we saw weird behavior not using it in the beginning of this project. I will double-check.

Reply:
Hi, below is response from MKL-DNN team:
"Using of shared_ptr in tests helps us to keep memory as class members. It allows to create memory not in constructor of test, but when we are ready to create it (by shared_ptr::reset) and also it automatically deallocates memory. Using of unique_ptr instead of shared_ptr should work as well."
Since shared_ptr has been used in TF (also in MklReluOp), I would suggest that we do not change it to unique_ptr.
Please let me know if you insist the change. Many thanks!